meterfs/meterfs.c

```diff
@@ -755,14 +756,19 @@ static int meterfs_writeRecord(file_t *f, const void *buff, size_t bufflen, mete
 	meterfs_powerCtrl(1, ctx);
 
 	/* Check if we have to erase sector to write new data */
-	if (offset == 0 || (offset / ctx->sectorsz) != ((offset + f->header.recordsz + sizeof(entry_t)) / ctx->sectorsz)) {
-		stat = ctx->eraseSector(ctx->devCtx, ctx->offset + (f->header.sector * ctx->sectorsz) + (offset + f->header.recordsz + sizeof(entry_t)));
-		if (stat < 0) {
-			meterfs_powerCtrl(0, ctx);
-			free(tdata);
-			return stat;
+	if (!f->earlyErased) {
+		if (offset == 0 || (offset / ctx->sectorsz) != ((offset + f->header.recordsz + sizeof(entry_t)) / ctx->sectorsz)) {
```

Comment:
IMHO we should clear `earlyErased` only if we enter this `if`

Reply:
Could you elaborate more? `earlyErased` is already cleared inside `if`.

Reply:
Hmm, OK, I guess it's OK as is, earlyErased is set true only right after we've erased the sector, so next write will always make use of it.

What I have in mind:
if (we should erase) {
 if (earlyErase) then don't erase and clear earlyErase
 else erase

but this is functionally equivalent because earlyErased is true only if we should erase